### PR TITLE
Add release notes for July release

### PR DESCRIPTION
**Pull request by markcowl**

### CODE REVIEW

File: docs/release-notes/release-2023-07-11.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- Add support for `UTF-8 with bom` for other files loaded by typespec compiler. `.tsp` files already had support, this make it more available for any library/emitter using the compiler api to load a file.
+- Allow annotations(Decorators, directives and doc comments) to be specified in any order
+- Format doc comments above decorators and directives
+- Improve formatting for properties with comments and decorators
+- Support for new Unicode 15.0 identifier characters
+
+### Emitter framework
+
+- Add `writeOutput` to TypeEmitter for additional customization in writign generated files
+- Add `meta`` property to source files and declarations to store arbitrary metadata
```
Comment (timotheeguerin) [inline docs/release-notes/release-2023-07-11.md:25]:
```suggestion
- Add `meta` property to source files and declarations to store arbitrary metadata
```

**Review by timotheeguerin — APPROVED** [review]

File: docs/release-notes/release-2023-07-11.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+:::danger
+This release contains **breaking changes**
+:::
+
+## New Features
+
+### `@typespec/compiler` and core language
+
+- New built-in linter system. Typespec libraries are able to define linting rules which can be configured in `tspconfig.yaml`. See documentation for configuring a [linter](https://microsoft.github.io/typespec/introduction/configuration#linter---configuring-linters) and [writing a linter](https://microsoft.github.io/typespec/extending-typespec/linters)
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:15]:
```suggestion
- New built-in linter system. Typespec libraries are able to define linting rules which can be configured in `tspconfig.yaml`. See documentation for [configuring a linter](https://microsoft.github.io/typespec/introduction/configuration#linter---configuring-linters) and [writing a linter](https://microsoft.github.io/typespec/extending-typespec/linters)
```

File: docs/release-notes/release-2023-07-11.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- Add support for `UTF-8 with bom` for other files loaded by typespec compiler. `.tsp` files already had support, this make it more available for any library/emitter using the compiler api to load a file.
+- Allow annotations(Decorators, directives and doc comments) to be specified in any order
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:17]:
```suggestion
- Allow decorators, directives and doc comments to be specified in any order
```

File: docs/release-notes/release-2023-07-11.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+:::danger
+This release contains **breaking changes**
+:::
+
+## New Features
+
+### `@typespec/compiler` and core language
+
+- New built-in linter system. Typespec libraries are able to define linting rules which can be configured in `tspconfig.yaml`. See documentation for configuring a [linter](https://microsoft.github.io/typespec/introduction/configuration#linter---configuring-linters) and [writing a linter](https://microsoft.github.io/typespec/extending-typespec/linters)
+- Add support for `UTF-8 with bom` for other files loaded by typespec compiler. `.tsp` files already had support, this make it more available for any library/emitter using the compiler api to load a file.
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:16]: This feels like a bugfix to me.

File: docs/release-notes/release-2023-07-11.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+- Add documentation on automatic header name resolution from property name for `@header` decorator
+
+### `@typespec/json-schema` emitter
+
+- Support `@extension` for adding arbitrary vendor extensions into the output
+- Add support for `Record<T>`
+- Support templates instantiated with intrinsic types and type expressions
+- Export emitter and related types from the npm package
+- By default, types that are not marked with @jsonSchema or are within a namespace with @jsonSchema are bundled into the schemas that reference them. Set the `emitAllRefs` option to true to get the previous behavior of emitting all types referenced as JSON Schema
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:39]:
```suggestion
- By default, types that are not marked with `@jsonSchema` or are within a namespace with `@jsonSchema` are bundled into the schemas that reference them. Set the `emitAllRefs` option to true to get the previous behavior of emitting all types referenced as JSON Schema
```

File: docs/release-notes/release-2023-07-11.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+- OpenAPI emitters: treat union namespaces like namespaces for other types
+
+### Emitter framework bug fixes
+
+- Fix that some context methods were not being passed the expected parameters
+- Fix that context was set incorrectly for some `TypeEmitter` methods, and add missing context methods for model properties, enum members, and union variants
+
+### `@typespec/json-schema` emitter bug fixes
+
+- Json Schema Emitter: Fix a bug that could result in a schema being bundled more than once
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:73]:
```suggestion
- Fix a bug that could result in a schema being bundled more than once
```

File: docs/release-notes/release-2023-07-11.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+## Breaking Changes
+
+### For Specification Authors
+
+- Update tsp init template schema for future extensibility. Older tsp version will fail validation when trying to initialize a project with a new template definition.
+- Json Schema Emitter: the namespace has been corrected to TypeSpec.JsonSchema
```
Comment (bterlson) [inline docs/release-notes/release-2023-07-11.md:80]:
```suggestion
- Json Schema Emitter: the namespace has been corrected to TypeSpec.JsonSchema. Update any using statements from `JsonSchema` to `TypeSpec.JsonSchema`, and any references from `JsonSchema.[Type]` to `TypeSpec.JsonSchema.[Type]`.
```

Might be a bit too detailed but I feel it's always good to say how to fix an error if you can.

**Review by bterlson — APPROVED** [review]

File: docs/release-notes/release-2023-07-11.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+- New built-in linter system. Typespec libraries are able to define linting rules which can be configured in `tspconfig.yaml`. See documentation for configuring a [linter](https://microsoft.github.io/typespec/introduction/configuration#linter---configuring-linters) and [writing a linter](https://microsoft.github.io/typespec/extending-typespec/linters)
+- Add support for `UTF-8 with bom` for other files loaded by typespec compiler. `.tsp` files already had support, this make it more available for any library/emitter using the compiler api to load a file.
+- Allow annotations(Decorators, directives and doc comments) to be specified in any order
+- Format doc comments above decorators and directives
+- Improve formatting for properties with comments and decorators
+- Support for new Unicode 15.0 identifier characters
+
+### Emitter framework
+
+- Add `writeOutput` to TypeEmitter for additional customization in writign generated files
```
Comment (tjprescott) [inline docs/release-notes/release-2023-07-11.md:24]:
```suggestion
- Add `writeOutput` to TypeEmitter for additional customization in writing generated files
```

**Pull request merged**